### PR TITLE
Added authorization and authentication

### PR DESCRIPTION
**Pull request by taltaub22**

### CODE REVIEW

File: routers/crudRouter.js [hunk 1]
```diff
@@ -2,6 +2,9 @@
 const express = require('express')
 const MongoClient = require('mongodb').MongoClient
 const MsLogger = require('@first-lego-league/ms-logger').Logger()
+const { authroizationMiddlware } = require('@first-lego-league/ms-auth')
+
+const adminAction = authroizationMiddlware(['admin', 'scorekeeper', 'development'])
```
Comment (idanstark42) [inline routers/crudRouter.js]: Notice that in the scoring app i allowed the scorekeeper to do anything that the admin could do. I don't think this rule should be in effect here. I think that `adminAction` shuold be only performed by `admin` or `development`

File: index.js [hunk 2]
```diff
@@ -2,8 +2,11 @@
 
 const express = require('express')
 const domain = require('domain')
-const {correlateSession} = require('@first-lego-league/ms-correlation')
-const msLogger = require('@first-lego-league/ms-logger').Logger()
+const { correlateSession } = require('@first-lego-league/ms-correlation')
```
Comment (idanstark42) [inline index.js]: You require this package twice. Here and in line 10 (previosly 7)

File: index.js [hunk 3]
```diff
@@ -16,14 +19,21 @@ const appPort = process.env.PORT || 3001
 
 const bodyParser = require('body-parser')
 
-msLogger.setLogLevel(process.env.LOG_LEVEL || msLogger.LOG_LEVELS.DEBUG)
+logger.setLogLevel(process.env.LOG_LEVEL || logger.LOG_LEVELS.DEBUG)
 
 const app = express()
 app.use(bodyParser.json())
 app.use(msCorrelation.correlationMiddleware)
+app.use(msLogger.middleware)
```
Comment (idanstark42) [inline index.js]: The logger middleware if called `loggerMiddleware`, and not just `middleware`

File: index.js [hunk 4]
```diff
@@ -16,14 +19,21 @@ const appPort = process.env.PORT || 3001
...
+
+if (process.env.DEV) {
+  app.use(authenticationDevMiddleware())
```
Comment (idanstark42) [inline index.js]: You're using the authentication on any request. This means that any request with a user that is not authenticated will be redirected, not just the UI.
I now notice that the UI doesn't exist yet in this PR. You can just put here a TODO until it is. Also, is all CRUD operations that the UI allows are only available to admins, maybe the entire UI should only be available to admins...

File: index.js (hunk 3 above)
Comment (taltaub22) [reply index.js]: That was working....

File: index.js (hunk 4 above)
Comment (taltaub22) [reply index.js]: Ok ,first of all, Who is responsible for the login ui?
Second, Thats why you saw the cors thing, It wont  be here after the ui would be served from the node server

File: index.js (hunk 3 above)
Comment (idanstark42) [reply index.js]: You didn't run `yarn upgrade`

File: index.js (hunk 4 above)
Comment (idanstark42) [reply index.js]: I'm responsible for the login UI.
I didn't get the second point. Could you explain?

Comment (taltaub22) [reply index.js]: The cors thing was in the other pr's because of the client.
If the node server is serving the client the cors is not needed any more.

Comment (idanstark42) [reply index.js]: You do need the cors because you need other services to send requests to your API, and different ports require CORS

**Pull request merged**